File: java/oop/StudentApp/src/com/techlabs/student/Student.java
package com.techlabs.student;

public class Student{
	private int rollno;
	private String name;
	private Integer standard;
	
	public Student(int rollno,String name,int standard){
		this.rollno=rollno;
		this.name=name;
		this.standard=standard;
	}
	
	public int getRollno(){
		return this.rollno;
	}
	
	public String getName(){
		return this.name;
	}
	
	public Integer getStandard(){
		return this.standard;
	}
	
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Student other = (Student) obj;
		if (rollno != other.rollno)
			return false;
		if (standard == null) {
			if (other.standard != null)
				return false;
		} else if (!standard.equals(other.standard))
			return false;
		return true;
	}
	
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + rollno;
		result = prime * result
				+ ((standard == null) ? 0 : standard.hashCode());
		return result;
	}
	
	@Override
	public String toString() {
		return " RollNo: "+this.rollno
				+" - Name: "+this.name
				+" - Standard: "+this.standard
				+" ";
	}
}

File: dotNet/CSharp/StudentApp/StudentApp/StudentClass/Student.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StudentApp.StudentClass
{
    public class Student
    {
        private int _rollNo;
        private string _name;
        private int _standard;

        public Student(int rollNo,string name,int standard)
        {
            this._rollNo = rollNo;
            this._name = name;
            this._standard = standard;
        }

        public Int32 RollNo
        {
            get
            {
                return this._rollNo;
            }
        }

        public String Name
        {
            get
            {
                return this._name;
            }
        }

        public Int32 Standard
        {
            get
            {
                return this._standard;
            }
        }

        public override bool Equals(object obj)
        {
            Student other = (Student)obj;

            if(other.RollNo == this.RollNo)
            {
                return false;
            }

            return true;
        }

        public override int GetHashCode()
        {
            return this.RollNo.GetHashCode();
        }

        public override string ToString()
        {
            return "Roll No: "+this.RollNo
                +" - Name: "+this.Name
                +" - Standard: "+this.Standard
                +" ";
        }
    }
}
